presto-product-tests/src/main/java/io/prestosql/tests/hive/TestAvroSymlinkInputFormat.java

```diff
@@ -87,12 +74,44 @@ public void testSymlinkTable()
                 "STORED AS " +
                 "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat' " +
                 "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'");
-
+        hdfsClient.createDirectory(warehouseDirectory + "/TestAvroSymlinkInputFormat/data");
+        saveResourceOnHdfs("avro/original_data.avro", warehouseDirectory + "/TestAvroSymlinkInputFormat/data/original_data.avro");
         hdfsClient.delete(warehouseDirectory + "/test_avro_schema_symlink/symlink.txt");
         hdfsClient.saveFile(warehouseDirectory + "/test_avro_symlink/symlink.txt", format("hdfs://%s/TestAvroSymlinkInputFormat/data/original_data.avro", warehouseDirectory));
-
-        assertThat(onPresto().executeQuery("SELECT * FROM test_avro_symlink"))
-                .containsExactly(row("someValue", 1));
+        assertThat(onPresto().executeQuery("SELECT * FROM test_avro_symlink")).containsExactly(row("someValue", 1));
         onHive().executeQuery("DROP TABLE IF EXISTS test_avro_symlink");
+        hdfsClient.delete(warehouseDirectory + "/TestAvroSymlinkInputFormat");
+    }
+
+    @Test(groups = {AVRO, STORAGE_FORMATS})
+    public void testSymlinkTableWithDirectory()
+            throws IOException
+    {
+        onHive().executeQuery("DROP TABLE IF EXISTS test_avro_symlink_with_directory");
+
+        onHive().executeQuery("" +
+                "CREATE TABLE test_avro_symlink_with_directory " +
+                "ROW FORMAT SERDE 'org.apache.hadoop.hive.serde2.avro.AvroSerDe' " +
+                "WITH SERDEPROPERTIES ('avro.schema.literal'='{" +
+                "\"namespace\": \"io.prestosql.tests.hive\"," +
+                "\"name\": \"test_avro_symlink\"," +
+                "\"type\": \"record\"," +
+                "\"fields\": [" +
+                "{ \"name\":\"string_col\", \"type\":\"string\"}," +
+                "{ \"name\":\"int_col\", \"type\":\"int\" }" +
+                "]}') " +
+                "STORED AS " +
+                "INPUTFORMAT 'org.apache.hadoop.hive.ql.io.SymlinkTextInputFormat' " +
+                "OUTPUTFORMAT 'org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat'");
+
+        String dataRoot = "/TestAvroSymlinkInputFormat";
+        String dataDir = dataRoot + "/data2/subdir";
+        hdfsClient.createDirectory(warehouseDirectory + dataDir);
+        saveResourceOnHdfs("avro/original_data.avro", warehouseDirectory + dataDir + "/original_data.avro");
```

Comment:
put some extra file in `/data2` and `/data2/subdir` directories, so that we are sure it gets skipped